Rules/CommonScripts/EmoteBinderMenu.as

```diff
@@ -0,0 +1,166 @@
+
+			//press escape to close
+			CBitStream params;
+			menu.AddKeyCommand(KEY_ESCAPE, rules.getCommandID("close menu"), params);
+			menu.SetDefaultCommand(rules.getCommandID("close menu"), params);
+
+			//display emote grid
+			for (int i = 0; i < Emotes::emotes_total; i++)
+			{
+				if (EXCLUDED_EMOTES.find(i) > -1)
+				{
+					continue;
+				}
+
+				CBitStream params;
+				params.write_u8(i);
+				CGridButton@ button = menu.AddButton(getIconName(i), description, rules.getCommandID("bind emote"), Vec2f(1, 1), params);
+			}
+			
+			//fill in extra slots in emote grid
+			menu.FillUpRow();
+
+			//separator with info
+			CGridButton@ separator = menu.AddTextButton("Select a keybind below, then select the emote you want", Vec2f(MENU_WIDTH, 1));
+			separator.clickable = false;
+			separator.SetEnabled(false);
+
+			//get current emote keybinds
+			ConfigFile cfg = ConfigFile();
+			if (!cfg.loadFile("../Cache/EmoteBindings.cfg") &&
+				!cfg.loadFile("EmoteBindings.cfg"))
+			{
+				return;
+			}
+
+			array<u8> emoteBinds = {
+				read_emote(cfg, "emote_1", Emotes::attn),
+				read_emote(cfg, "emote_2", Emotes::smile),
+				read_emote(cfg, "emote_3", Emotes::frown),
+				read_emote(cfg, "emote_4", Emotes::mad),
+				read_emote(cfg, "emote_5", Emotes::laugh),
+				read_emote(cfg, "emote_6", Emotes::wat),
+				read_emote(cfg, "emote_7", Emotes::troll),
+				read_emote(cfg, "emote_8", Emotes::disappoint),
+				read_emote(cfg, "emote_9", Emotes::ladder)
+			};
+			
+			//display row of current emote keybinds
+			for (int i = 0; i < 9; i++)
+			{
+				CBitStream params;
+				params.write_u8(i);
+				CGridButton@ button = menu.AddButton(getIconName(emoteBinds[i]), "Select key " + (i + 1), rules.getCommandID("select keybind"), Vec2f(1, 1), params);
+				button.selectOneOnClick = true;
+				// button.hoverText = "     Key " + (i + 1) + "\n";
+
+				//reselect keybind if one was selected before
+				if (selected == i)
+				{
+					button.SetSelected(1);
+				}
+			}
+		}
+	}
+}
+
+void onCommand(CRules@ this, u8 cmd, CBitStream @params)
+{
+	if (cmd == this.getCommandID("bind emote"))
+	{
+		//must select keybind first
+		if (selected == -1)
+		{
+			return;
+		}
+
+		u8 emote = params.read_u8();
+		string key = "emote_" + (selected + 1);
+
+		//get emote bindings cfg file
+		ConfigFile cfg = ConfigFile();
+		if (!cfg.loadFile("../Cache/EmoteBindings.cfg") &&
+			!cfg.loadFile("EmoteBindings.cfg"))
+		{
+			return;
+		}
+
+		//bind emote
+		cfg.add_string(key, "" + emote);
+		cfg.saveFile("EmoteBindings.cfg");
+
+		//update keybinds in menu
+		ShowMenu();
+	}
+	else if (cmd == this.getCommandID("select keybind"))
+	{
+		selected = params.read_u8();
+	}
+	else if (cmd == this.getCommandID("close menu"))
+	{
+		getHUD().ClearMenus(true);
+	}
+}
+
+void onMainMenuCreated(CRules@ this, CContextMenu@ menu)
+{
+	//add button to main menu
+	Menu::addContextItem(menu, getTranslatedString("Bind Emotes"), "EmoteBinderMenu.as", "void NewMenu()");
```

Comment:
Maybe have a "bindings" parent menu for this so more things like that can be added in the same submenu? e.g. taunts